test/testdevice.py

```diff
@@ -21,8 +21,10 @@ def has_audio_recording(self):
             self.devicename))
         return self.upload_audio_recording()
 
-    def upload_recording(self):
+    def upload_recording(self, properties = {}):
```

Comment:
Mutable objects as keyword arg defaults are dangerous. See https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

It looks like the way you're using it doesn't suffer from the problem but it's best to avoid this kind of thing. Set the default as None and handle that in the function.